### PR TITLE
Issue 148

### PR DESCRIPTION
**Pull request by fabiomaffioletti**

### CODE REVIEW

File: jsondoc-ui-webjar/src/main/resources/jsondoc-ui.html [hunk 1]
```diff
@@ -586,6 +589,23 @@ <h4 class="panel-title">
...
+							<option value="{{this}}">{{this}}</option>
+						{{/each}}
+					</select>
+				</div>
+				<div class="form-group" style="margin-bottom:5px;">
+					<label for="tokenAuthToken">Token</label>
+					<input class="form-control" type="text" id="tokenAuthToken" name="tokenAuthToken" placeholder="Token">
+				</div>
+			</div>
+		{{/equal}}
```
Comment (strokine) [inline jsondoc-ui-webjar/src/main/resources/jsondoc-ui.html]: Change suggestion:
```
{{#equal auth.type "TOKEN"}}
			<div class="col-md-12">
				<h4>Token Authentication</h4>
				{{#if auth.testtokens}}
				<div class="form-group">
					<select class="form-control" id="tokenAuthSelect" onchange="fillTokenAuthFields(); return false;">
						<option disabled="disabled" selected="selected">Select a test token or fill input below</option>
						{{#each auth.testtokens}}
							<option value="{{this}}">{{this}}</option>
						{{/each}}
					</select>
				</div>
				{{/if}}
				<div class="form-group" style="margin-bottom:5px;">
					{{#if auth.testtokens}}
					<label for="tokenAuthToken">Token</label>
					{{/if}}
					<input class="form-control" type="text" id="tokenAuthToken" name="tokenAuthToken" placeholder="Token">
				</div>
			</div>
		{{/equal}}
```

File: jsondoc-ui-webjar/src/main/resources/jsondoc-ui.html [hunk 2]
```diff
@@ -1016,6 +1040,14 @@ <h5 style="padding:0px">Request URL</p>
 							if(method.auth.type == "BASIC_AUTH") {
 								headers["Authorization"] = "Basic " + window.btoa($('#basicAuthUsername').val() + ":" + $('#basicAuthPassword').val());
 							}
+							
+							if(method.auth.type == "TOKEN") {
+								if(method.auth.scheme != "") {
```
Comment (strokine) [inline jsondoc-ui-webjar/src/main/resources/jsondoc-ui.html]: If scheme is set to "" servlet doesn't return the field. Need it fix this line with:
```
if(typeof(method.auth.scheme) != "undefined" && method.auth.scheme != "") {
```

File: jsondoc-core/src/main/java/org/jsondoc/core/annotation/ApiAuthToken.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+@Documented
+@Target(value = { ElementType.TYPE, ElementType.METHOD})
+@Retention(RetentionPolicy.RUNTIME)
+public @interface ApiAuthToken {
+	
+	/**
+	 * The role(s) a user must have to correctly access the annotated controller/method
+	 * @return
+	 */
+	public String[] roles();
```
Comment (strokine) [inline jsondoc-core/src/main/java/org/jsondoc/core/annotation/ApiAuthToken.java]: I'd suggest:
```
public String[] roles() default {};
```

**Comment by fabiomaffioletti** [comment]
@strokine thank you for your suggestions, I implemented them and released them right now. Instead of having an empty array for roles, I decided to put ```{"*"}``` as default value. Available in a few hours on maven central.

**Pull request merged**